crates/chain/src/tx_graph.rs

```diff
@@ -687,8 +701,20 @@ impl<A: Anchor> TxGraph<A> {
 
     /// Get the position of the transaction in `chain` with tip `chain_tip`.
     ///
-    /// If the given transaction of `txid` does not exist in the chain of `chain_tip`, `None` is
-    /// returned.
+    /// If the given `txid` does not exist in the chain of `chain_tip`, and
+    /// we believe that the transaction is not in mempool anymore, `None` is returned.
+    ///
+    /// We determine whether the transaction `tx` is in the mempool or not by looking at
+    /// tx's conflicting transactions, and tx's ancestors' conflicting transactions;
+    /// if in these transactions we find one which is anchored to a block in the best chain,
+    /// then `tx` can't be in mempool.
+    ///
+    /// Otherwise, we decide which transactions are in mempool based on their
+    /// `last_seen_unconfirmed`. We first of all calculate `tx`'s `max_last_seen_unconfirmed`,
+    /// which is the max `last_seen_unconfirmed` between `tx` and all its descendants.
+    /// We then look at all the conflicts of `tx`, and if all of them have
+    /// `last_seen_unconfirmed` < `max_last_seen_unconfirmed`, then we consider `tx` to be still
+    /// in mempool.
```

Comment:
Would it be better if we explain it like so?

> There are two factors that we use to approximate whether an unconfirmed transaction still exists in the mempool...
> 1. Conflicts with confirmed txs!
> 2. `last_seen`! We prioritize later `last_seen`s because...

Reply:
Yes but without exclamation marks :P

Also probably not "still exists" but instead "currently exists".

Reply:
That's better for sure, but I still think that the point 2 (`last_seen`) is not really clear. I'm refactoring this piece of doc as:
```
    /// If the given `txid` does not exist in the chain of `chain_tip`, and
    /// we believe that the transaction is not in mempool anymore, `None` is returned.
    ///
    /// There are two factors that we use to approximate whether an unconfirmed
    /// transaction currently exists in mempool:
    /// 1. Conflicts with confirmed txs. If the transaction `tx` or any of its unconfirmed
    /// ancestors conflict with a confirmed transaction, then `tx` can't be in mempool.
    /// 2. The `last_seen_unconfirmed` parameter. Given two conflicting transactions, we say
    /// that the one with a higher `last_seen_unconfirmed` (i.e., we saw it later) has a higher
    /// chance of still being in mempool. When trying to figure out if `tx`
    /// is still in mempool, we first of all calculate `tx`'s `max_last_seen_unconfirmed`,
    /// which is the max `last_seen_unconfirmed` between `tx` and all its descendants.
    /// We then look at all the conflicts of `tx`, and if for all of them
    /// `last_seen_unconfirmed` < `max_last_seen_unconfirmed` holds,
    /// then we consider `tx` to be still in mempool.
```

How would you rephrase it?

Reply:
```
/// Get the position of the transaction in a chain with tip `chain_tip`.
///
/// Chain data is fetched from a [`ChainOracle`] implementation - `chain`.
///
/// This method returns `Ok(None)` if the transaction is not found in the chain, and no longer
/// belongs in the mempool. The following factors are used to approximate whether an 
/// unconfirmed transaction exists in the mempool (not evicted):
///
/// 1. Unconfirmed transactions that conflict with confirmed transactions are evicted.
/// 2. Unconfirmed transactions that spend from transactions that are evicted, are also
///    evicted.
/// 3. Given two conflicting unconfirmed transactions, the transaction with the lower
///    `last_seen_unconfirmed` parameter is evicted. A transaction's `last_seen_unconfirmed` 
///    parameter is the max of all it's descendants' `last_seen_unconfirmed` parameters. If the
///    final `last_seen_unconfirmed`s are the same, the transaction with the lower `txid` (by
///    lexicographical order) is evicted.
///
/// # Error
///
/// An error will occur if the [`ChainOracle`] implementation (`chain`) fails. If the
/// [`ChainOracle`] is infallible, [`get_chain_position`] can be used instead.
///
/// [`get_chain_position`]: Self::get_chain_position
```

@danielabrozzoni What do you think about this?

Reply:
Looks good! Updated, thanks